src/OrbitCore/SymbolHelper.cpp

```diff
@@ -126,25 +132,59 @@ std::vector<fs::path> FindStructuredDebugDirectories() {
 
 }  // namespace
 
-ErrorMessageOr<void> SymbolHelper::VerifySymbolsFile(const fs::path& symbols_path,
-                                                     const std::string& build_id) {
-  OUTCOME_TRY(symbols_file, CreateElfFile(symbols_path));
+ErrorMessageOr<void> SymbolHelper::VerifySymbolsFile(
+    const fs::path& symbols_path,
+    llvm::object::OwningBinary<llvm::object::ObjectFile>&& symbols_file,
+    std::string_view build_id) {
+  const auto* binary = symbols_file.getBinary();
 
-  if (!symbols_file->HasDebugSymbols()) {
-    return ErrorMessage(
-        absl::StrFormat("Elf file \"%s\" does not contain symbols.", symbols_path.string()));
+  if (binary->isELF()) {
```

Comment:
this could be member of Orbit ObjectFile, I feel like there is too much of LLVM leaking outside of implementation of Elf/Coff files. Seems like part of this code could be incapsulated into a more generic factory method.

Reply:
Yeah, I agree. I have been working on a  better (?) version that creates generic ObjectFiles etc. and only casts to Elf or Coff when needed. I'll commit that to see what you think about it. In particular, it removes the additional dependency of llvm in Service.

Reply:
Done.